src/amplitude.js

```diff
@@ -433,33 +552,45 @@ Amplitude.prototype.setUserId = function(userId) {
   try {
     this.options.userId = (userId !== undefined && userId !== null && ('' + userId)) || null;
     _saveCookieData(this);
-    // utils.log('set userId=' + userId);
   } catch (e) {
     utils.log(e);
   }
 };
 
-Amplitude.prototype.setOptOut = function(enable) {
-  if (!this._apiKeySet('setOptOut()')) {
+/**
+ * Sets whether to opt current user out of tracking.
+ * @public
+ * @param {boolean} enable - if true then no events will be logged or sent.
+ * @example: amplitude.setOptOut(true);
+ */
+Amplitude.prototype.setOptOut = function setOptOut(enable) {
+  if (!this._apiKeySet('setOptOut()') || !utils.validateInput(enable, 'enable', 'boolean')) {
```

Comment:
Actually I might need to rethink this. In v1 of multiple apps, each instance had its own cookie to save different config settings. The way we name the cookies is amplitude_ first 6 chars of the api key. For that reason, any public functions that logged an event, or interacted with the cookie in any way (modifying/saving/loading) would need the apiKey set. In v2 we actually namespace on the instance name instead of the apiKey, so it actually might not be necessary to always check for the apiKey now. Of course any of the user property stuff (like clearUserProperties) would still need the check since they trigger an identify event, which will post a request.